### PR TITLE
COL-673: Get rid of either `name` or `channelName` in @ably-labs/models

### PR DESCRIPTION
**Pull request by snikidev**

### CODE REVIEW

File: src/types/model.ts [hunk 1]
```diff
@@ -58,7 +58,8 @@ export type ModelsOptions = OptionalValues<
  * @interface
  */
 export type ModelSpec<S extends SyncFuncConstraint> = {
-  name: string;
+  // LEGACY: we're phasing out the use of name in favour of channelName
+  name?: string;
```
Comment (mschristensen) [inline src/types/model.ts:62]: Let's just remove this entirely - what else depends on it that's not updated in this PR?

Comment (snikidev) [reply src/types/model.ts:62]: That's a breaking change. If someone uses our SDK already, then their linter would go crazy and they'd need to refactor all their instances when updating. we're okay with that?

Comment (mschristensen) [reply src/types/model.ts:62]: Yes I'm happy to remove this, we will just bump the release tag according to semver :)

File: src/ModelsClient.ts [hunk 2]
```diff
@@ -89,19 +89,18 @@ export default class ModelsClient {
           throw new InvalidArgumentError('expected spec to be a ModelSpec');
         }
 
-        const name = spec.name;
         const channelName = spec.channelName;
 
-        if (!name) {
+        if (!channelName) {
           throw new InvalidArgumentError('Model must have a non-empty name');
```
Comment (mschristensen) [inline src/ModelsClient.ts:95]: Should say "non-empty channel name"

**Review by mschristensen — CHANGES_REQUESTED** [review]
Let's remove the legacy name option and then I'm happy to merge this

**Review by mschristensen — APPROVED** [review]

**Pull request merged**